### PR TITLE
Added documentation for React integration

### PR DESCRIPTION
**Pull request by jameshibbard**
Adds docs on React integration (as discussed [here](https://github.com/shakacode/shakapacker/issues/92#issuecomment-1101227999)).

### CODE REVIEW

File: docs/react.md [hunk 1]
```diff
@@ -0,0 +1,262 @@
+# React Integration
+
+These steps describe how to create a Rails/React app, using Shakapacker as the bundler.
+
+Before starting, ensure that you have Yarn installed:
+
+```shell
+npm i -g yarn
```
Comment (justin808) [inline docs/react.md:8]: `asdf` and `yvm` are other ways.

Let's say, "for example"

File: docs/react.md [hunk 2]
```diff
@@ -0,0 +1,262 @@
...
+}
+```
+
+9. Start the Rails server and the webpack-dev-server in separate console windows:
+```shell
+rails s
+./bin/webpacker-dev-server
+```
+
+10. Hit: <http://localhost:3000/site/index>
```
Comment (justin808) [inline docs/react.md:258]: Is using `<URL>` better? Do we need the `<`?

**Review by justin808 — CHANGES_REQUESTED** [review]
Exciting. A couple suggestions!

File: docs/react.md (hunk 1 above)
Comment (jameshibbard) [reply docs/react.md:8]: Sure. I updated the PR.

File: docs/react.md (hunk 2 above)
Comment (jameshibbard) [reply docs/react.md:258]: It's a shortcut style for creating “automatic” links for URLs, which means you can show the text of the URL and also have it be a clickable link. See [here](https://daringfireball.net/projects/markdown/syntax#autolink). 

Happy to remove if it's not needed.

**Review by tomdracz — APPROVED** [review]
LGTM, nice work @jameshibbard

**Pull request merged**